### PR TITLE
[store_adapter] Adapt Partitioner for vineyard on Read Path

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

As titled

## Related issue number

#501

### CODE REVIEW

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs [hunk 1]
```diff
@@ -14,13 +14,17 @@
...
-use gremlin_core::{str_to_dyn_error, DynResult, Partitioner, ID, ID_MASK};
 use maxgraph_store::api::graph_partition::GraphPartitionManager;
-use maxgraph_store::api::{Edge, GlobalGraphQuery, PropId, Vertex, VertexId};
+use maxgraph_store::api::{Edge, GlobalGraphQuery, Vertex};
 use std::sync::Arc;
+use crate::graph_proxy::partitioner::{MaxGraphMultiPartition, VineyardMultiPartition};
+use std::collections::HashMap;
 
+/// Initialize GremlinJobCompiler for v2
 pub fn initialize_job_compiler<V, VI, E, EI>(
```
Comment (longbinlai) [inline research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs:28]: Please merge these two function into one single trait, with the function `initialize_job_compiler()`

Comment (BingqingLyu) [reply research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs]: Done

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs [hunk 2]
```diff
@@ -14,63 +14,94 @@
...
+        MaxGraphStorage {
+            graph_query,
+            graph_partitioner,
+            num_servers,
+            server_index,
+        }
+    }
+}
+
+impl<V, VI, E, EI> InitializeJobCompiler for MaxGraphStorage<V, VI, E, EI>
```
Comment (longbinlai) [inline research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs:54]: MaxGraphStorage better be QueryMaxGraph

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs [hunk 3]
```diff
@@ -14,63 +14,94 @@
...
-        let server_index = self
-            .graph_partition_manager
-            .get_server_id(partition_id as PropId)
-            .ok_or(str_to_dyn_error(
-                "get server id failed in graph_partition_manager",
-            ))? as u64;
-        let worker_index = partition_id % worker_num_per_server;
-        Ok(server_index * worker_num_per_server + worker_index as u64)
+/// Initialize GremlinJobCompiler for vineyard
+impl<V, VI, E, EI> InitializeJobCompiler for VineyardStorage<V, VI, E, EI>
```
Comment (longbinlai) [inline research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs:95]: VineyardStorage -> QueryVineyard.

Comment (BingqingLyu) [reply research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs]: Done

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs (hunk 2 above)
Comment (BingqingLyu) [reply research/gaia/gremlin/gs_gremlin/src/graph_proxy/mod.rs]: Done

**Review by longbinlai — APPROVED** [review]

**Pull request merged**